crates/test-utils/src/sui_system_state.rs

```diff
@@ -90,5 +91,6 @@ pub fn test_sui_system_state(epoch: EpochId, validators: Vec<Validator>) -> SuiS
         },
         reference_gas_price: 1,
         validator_report_records: VecMap { contents: vec![] },
+        chain_id: TransactionDigest::genesis(),
```

Comment:
I think the hard part of `chain_id` will be making this function pluggable rather than pure. Not sure if the best way is an enum, Cargo flags or something else-- @bmwill may have ideas here.

Reply:
why should it be something pluggable instead of using the calculated digest from here? `let genesis_digest = genesis_ctx.digest();`?